### PR TITLE
UI interface

### PR DESCRIPTION
**Pull request by anyepeter**
Hello 😁👋, In this PR, my partners and I implement a catalog app with the following requirements📑;

### General requirements
✅ Make sure that there are no linter errors.
✅ Make sure that you used correct Gitflow.
✅ Make sure that you documented your work in a professional way.
### Ruby requirements
✅ Follow our list of best practices for Ruby.
#### Logic
✅ Each class from the UML class diagram should be created in a separate .rb file.
✅ All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
✅ All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
✅ All methods visible in the diagram should be implemented:
 - add_item method (in all classes that have that method)
     ✅ should take an instance of the Item class as an input.
     ✅ should add the input item to the collection of items.
     ✅ should add self as a property of the item object (by using the correct setter from the item object).
 - can_be_archived?() in the Item class
     ✅ should return true if published_date is older than 10 years.
     ✅ otherwise, it should return false.
- move_to_archive() in the Item class
     ✅ should reuse can_be_archived?() method.
     ✅ should change the archived property to true if the result of the can_be_archived?() method is true.
     ✅ should do nothing if the result of the can_be_archived?() method is false.
- can_be_archived?() in the Book class
     ✅ should override the method from the parent class.
     ✅ should return true if parent's method returns true OR if cover_state equals to "bad".
     ✅ otherwise, it should return false.
- can_be_archived?() in the MusicAlbum class
     ✅ should override the method from the parent class.
     ✅ should return true if parent's method returns true AND if on_spotify equals true.
     ✅ otherwise, it should return false.
- can_be_archived?() in the Movie class
     ✅ should override the method from the parent class.
     ✅ should return true if parent's method returns true OR if silent equals true.
     ✅ otherwise, it should return false.
- can_be_archived?() in the Game class
     ✅ should override the method from the parent class.
     ✅ should return true if parent's method returns true AND if last_played_at is older than 2 years.
     ✅ otherwise, it should return false.
- Add unit tests for all implemented methods.
#### User interface
✅ Create a main.rb file that will serve as your console app entry-point.
- Your console app, at the start, should:
   ✅ Present the user with a list of options to perform.
   ✅ Let users choose an option.
   ✅ If needed, ask for parameters for the option.
   ✅ Have a way to quit the app.
- The following options should be available:
   ✅ List all books
   ✅ List all music albums
   ✅ List all movies
   ✅ List of games
   ✅ List all genres (e.g 'Comedy', 'Thriller')
   ✅ List all labels (e.g. 'Gift', 'New')
   ✅ List all authors (e.g. 'Stephen King')
   ✅ List all sources (e.g. 'From a friend', 'Online shop')
   ✅ Add a book
   ✅ Add a music album
   ✅ Add a movie
   ✅ Add a game
✅All data should be preserved by saving collections in .json files.
#### Database
✅ Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.

capstone

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,161 +60,103 @@ After you're finished please remove all the comments and instructions!
```
Comment (fobadara) [inline README.md:22]: [REQUIRED] Kindly remove the microverse template logo and you can optionally replace it with your logo

File: README.md [hunk 2]
```diff
@@ -60,161 +60,103 @@ After you're finished please remove all the comments and instructions!
...
-- **[key_feature_1]**
-- **[key_feature_2]**
-- **[key_feature_3]**
+- **[add books]**
+- **[add moves]**
+- **[add games]**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- LIVE DEMO -->
```
Comment (fobadara) [inline README.md:46]: [REQUIRED] Kindly include the link to a video describing your project as instructed

File: app/item.rb [hunk 3]
```diff
@@ -1,42 +1,41 @@
 require 'date'
```
Comment (fobadara) [inline app/item.rb:2]: [REQUIRED] Kindly use attr_reader here. This makes your code safer by granting only read access
```
```ruby
 class Item
    attr_reader :genre
    attr_reader :author
    attr_reader :label
    attr_reader :source # only if group has 4 members

    ...

    def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end
      ...
```

File: main.rb [hunk 4]
```diff
@@ -1,31 +1,95 @@
...
+  def load_data
+    load_books
+    load_sources
+    load_movie
+    load_music_album
+    load_author
+    load_genre
+    load_games
+    load_movie
+  end
```
Comment (fobadara) [inline main.rb:91]: [REQUIRED] Kindly account for when an app is loaded without any data in the json files. For example, when I delete the content of game.json and I load the app I get this error

```

/Library/Ruby/Gems/2.6.0/gems/json-2.6.3/lib/json/common.rb:216:in `parse': unexpected token at '' (JSON::ParserError)
        from /Library/Ruby/Gems/2.6.0/gems/json-2.6.3/lib/json/common.rb:216:in `parse'
        from /Users/mac/Public/microverse-reviews/catalog-of-my-things/logic/game_logic.rb:20:in `load_games'
        from /Users/mac/Public/microverse-reviews/catalog-of-my-things/app.rb:13:in `initialize'
        from main.rb:6:in `new'
        from main.rb:6:in `application'
        from main.rb:95:in `<main>'
```

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Project looks professional
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**